### PR TITLE
Add database create script

### PR DESCRIPTION
**Pull request by YaleChen299**
Please help to see if any refactoring is possible! 

Will add tests soon if the structure is Okay.

### CODE REVIEW

File: internal/database/create.go
Comment (RichDom2185) [inline internal/database/create.go:19]: Let's use `logrus`, our logging framework, instead of `fmt`.

File: internal/config/config.go
Comment (RichDom2185) [inline internal/config/config.go:55]: I think you meant `constants.DB_DEFAULT_TIMEZONE` instead of name; either way, once #38 is merged, we can use `getEnvOrDefault` instead.

File: internal/utils/constants/db.go
Comment (RichDom2185) [inline internal/utils/constants/db.go:10]: Nice abstraction! But I don't think there's any "getting" involved here – would it be better if the function is simply named `OrDefault`?

_(Of course, it won't be used right now due to my above comment)_

File: internal/utils/constants/db.go
Comment (RichDom2185) [inline internal/utils/constants/db.go:7]: Thanks for catching this!

File: Makefile
Comment (RichDom2185) [inline Makefile:32]: We should add the targets to `DB_TARGETS` at the top of this file as well. That way, they'll be `.PHONY`-ed, and we can always set configurations that apply to all DB targets if we want to.

File: internal/database/connect.go
Comment (RichDom2185) [inline internal/database/connect.go:8]: To me, moving the DSN generation outside seems to initially break abstraction, since this DSN is specific to postgres.

Was there a reason this change was done? Was it testability or something else?

File: internal/database/create.go
Comment (RichDom2185) [reply internal/database/create.go:19]: Perhaps, `logrus.InfoLn`?

**Review by RichDom2185 — COMMENTED** [review]
Is there a way we don't have to depend on the internal `db` package to generate the connection?

File: internal/database/connect.go
Comment (YaleChen299) [reply internal/database/connect.go:8]: Right, I was thinking about this as well. Essentially, I need another function that connects to just the db server, since the database with the given name may not exist in the db server.

As you mentioned earlier, I could write another similar function and move all the create functionalities to the script folder?

File: internal/config/config.go
Comment (YaleChen299) [reply internal/config/config.go:55]: thanks for catching this!

**Review by RichDom2185 — CHANGES_REQUESTED** [review]
As discussed offline, to keep each of the functionalities pure, let's not automatically add the side effect of creating the DB if doesn't exist.

Instead, when running migrate/rollback/status scripts, it should just show an error if the database has yet to be created.

File: internal/config/db.go
Comment (RichDom2185) [inline internal/config/db.go:25]: I don't think this is the right approach. We already have a check that `c.DatabaseName` is not empty.

If we want to override the config, just create a new struct with the empty field:

```golang
oldConfig := conf.Database

// Copy over existing field values to a new struct
newConfig := config.DatabaseConfig(oldConfig)

// Set database name to empty
newConfig.DatabaseName = ""
```

File: Makefile
Comment (RichDom2185) [inline Makefile:4]: Does it not run if you don't include the second file?

File: internal/config/db.go
Comment (RichDom2185) [inline internal/config/db.go:52]: Refer to my comment above on why this condition is not necessary.

**Review by RichDom2185 — APPROVED** [review]
LGTM, feel free to merge if you agree with the changes. Thanks for working on this!

**Comment by YaleChen299** [comment]
> LGTM, feel free to merge if you agree with the changes. Thanks for working on this!

Thank you!

**Pull request merged**
